### PR TITLE
Include a full OS Login SSH Key example.

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7686

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by upodroid** [comment]
@melinath ^

File: third_party/terraform/tests/resource_os_login_ssh_public_key_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+resource "time_offset" "expiry" {
+  offset_hours = 1
+}
+
+resource "google_os_login_ssh_public_key" "cache" {
+  project = google_project.project.project_id
+  user    =  data.google_client_openid_userinfo.me.email
+  key     = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIPM4pxpbPpjuBocS6qlW0BHRYgH5xmv/yVrANZR9lc1N"
+  expiration_time_usec = time_offset.expiry.unix * 1000000
+  depends_on = [
```
Comment (melinath) [inline third_party/terraform/tests/resource_os_login_ssh_public_key_test.go:70]: Ideally this test would be as simple as possible to cover the functionality it's testing – would it be possible to remove these dependencies from the test configuration, or would that make it stop working?

**Review by melinath — CHANGES_REQUESTED** [review]
Hi @upodroid - thanks for the PR! I am realizing looking at the ticket I opened that I focused on the tests and forgot to talk about adding a docs example. The goal for this work would be to have a full docs example that is also used to auto-generate a full test.

Sorry about that!

Would it be possible to rewrite this as an example that then auto-generates a test?

**Comment by upodroid** [comment]
I won't be able to convert it to an auto generated test because I need to use the time provider which is loaded by the ExternalProviders field in the testcase. If you know how do it via MM do let me know and I'll make the changes.

File: third_party/terraform/tests/resource_os_login_ssh_public_key_test.go (hunk 1 above)
Comment (upodroid) [reply third_party/terraform/tests/resource_os_login_ssh_public_key_test.go:70]: The time has to be in microseconds or the API call fails.

Comment (upodroid) [reply third_party/terraform/tests/resource_os_login_ssh_public_key_test.go:70]: If you don't need project field being set, I can eliminate the resources creating the project

**Review by melinath — APPROVED** [review]
Hey @upodroid! After looking into this a bit it doesn't seem like it's worth it to convert this to a generated test. Sorry for the back and forth; I'm new to the team and wasn't quite sure about the best practice in this case. :-) Thanks for your contribution! The test LGTM - if you have time to create an example to go along with this (that's marked not to generate a test) that would be 💯 , otherwise I'll make a new ticket to track that work!

**Comment by upodroid** [comment]
That is fine. Personally I don't think we need it documented. Take GCE Instance for example https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance one of the most popular terraform resources. There is only one basic example in the docs but we have tons of tests in the background.

I might extend MM to hide the examples from the docs.

**Comment by melinath** [comment]
I ran the new test locally without issue; merging.

**Pull request merged**